packages/@pollyjs/adapter-node-http/src/utils/transport-wrapper.js

```diff
@@ -0,0 +1,212 @@
+import http from 'http';
+import zlib from 'zlib';
+
+import { URL } from '@pollyjs/utils';
+
+const NATIVE_END = Symbol();
+const LISTENERS = Symbol();
+
+const nativeRequestMapping = new WeakMap();
+
+export default class TransportWrapper {
+  constructor(transport, { name, adapter }) {
+    this.name = name;
+    this.adapter = adapter;
+    this.transport = transport;
+  }
+
+  isPatched() {
+    return nativeRequestMapping.has(this.transport);
+  }
+
+  patch() {
+    // make sure it's not already patched
+    this.adapter.assert(
+      `The ${
+        this.name
+      } transport has already been patched, please stop any running Polly instances`,
+      !this.isPatched()
+    );
+
+    nativeRequestMapping.set(this.transport, this.transport.request);
+    this.transport.request = this.createRequestWrapper();
```

Comment:
Do we also need to patch `request.get`?

Reply:
We should cover `http.get` with patching `request`:
```js
function get(url, options, cb) {
  var req = request(url, options, cb);
  req.end();
  return req;
}
```
It didn't seem to change from [2015](https://github.com/nodejs/node/blob/9cd44bb2b683446531306bbcff8739fc3526d02c/lib/http.js#L34) till [now](https://github.com/nodejs/node/blob/7cc3b1eac56d6a5759a298666ad11436396fdf0c/lib/http.js#L44)